pkg/reconciler/ingress/ingress_test.go

```diff
@@ -390,13 +363,44 @@ func TestReconcile(t *testing.T) {
 			Eventf(corev1.EventTypeNormal, "Updated", "Updated status for Ingress %q", "reconcile-virtualservice"),
 		},
 		Key: "test-ns/reconcile-virtualservice",
+	}, {
+		Name: "steady state",
+		Key:  "test-ns/route-tests",
+		Objects: []runtime.Object{
+			ingressWithStatus("route-tests", 1234, ingressReady),
+			resources.MakeMeshVirtualService(insertProbe(ingress("route-tests", 1234))),
+			resources.MakeIngressVirtualService(insertProbe(ingress("route-tests", 1234)),
+				makeGatewayMap([]string{"knative-testing/knative-test-gateway", "knative-testing/knative-ingress-gateway"}, nil)),
+			route(ingress("route-tests", 1234), "domain.com"),
+		},
+	}, {
+		Name:                    "remove wrong route",
```

Comment:
Can we come up with some more unit tests?

- Remove all routes after ingress is removed
- Create multiple routes

Reply:
> Remove all routes after ingress is removed

Ahhh. This is nice catch. We have this bug. (I guess that k-o-i also has it.) If we delete only ingress, new route is created by old one is not removed.

```
$ oc get routes -n knative-serving-ingress
NAME                                                      HOST/PORT                                                                                  PATH   SERVICES               PORT    TERMINATION   WILDCARD
route-09e4fe21-1757-11ea-bc17-0afc174df830-663366306138   HostAlreadyClaimed                                                                                istio-ingressgateway   http2   edge/Allow    None
route-0c6d0134-1756-11ea-bc17-0afc174df830-663366306138   hello-example.default.apps.knakayam-dev3.devcluster.openshift.com                                 istio-ingressgateway   http2   edge/Allow    None
```

Reply:
networking-openshift(as well as k-o-i) fails to create new route due to `already exists` error, so it breaks the loop before deleting old route :weary:

`failed to reconcile routes: failed to reconcile route: error creating route \"route-09e4fe21-1757-11ea-bc17-0afc174df830-663366306138\" for host \"hello-example.default.apps.knakayam-dev3.example.com\": routes.route.openshift.io \"route-09e4fe21-1757-11ea-bc17-0afc174df830-663366306138\" already exists"`

Reply:
Added owner reference and solved it.